File: CPP/Motorcycle.cpp
// Pre-processors and libraries
#pragma once
#include <string>
#include "Vehicle.cpp"

using namespace std;
        
// Motorcycle class
class Motorcycle : public Vehicle {
    private:
        // Attributes
        string bike_type;
        int tank_capacity;

    public:
        /* Constructors */

        // Empty Constructor
        Motorcycle() {
            this->bike_type = "n/a";
            this->tank_capacity = -1;
        }
        // Non-Empty Constructor
        Motorcycle(int bike_type, int tank_capacity, string license_plate, string brand, int prod_year, string color) : Vehicle(license_plate, brand, prod_year, color){
            this->bike_type = bike_type;
            this->tank_capacity = tank_capacity;
        }

        /* Methods */

        // Get Methods
        string getBikeType() { return this->bike_type; }
        int getTankCapacity() { return this->tank_capacity; }

        // Set Methods
        void setBikeType(string bike_type) { this->bike_type = bike_type; }
        void setTankCapacity(int tank_capacity) { this->tank_capacity = tank_capacity; }
};

File: Python/Motorcycle.py
from Vehicle import Vehicle # Imports Vehicle Class

class Motorcycle(Vehicle):
    # Attribute Declaration
    __bike_type = "";
    __tank_capacity = -1;


    # Constructor
    def __init__(self, bike_type=-1, tank_capacity=-1, license_plate="n/a", brand="n/a", prod_year=-1, color="n/a"):
        super().__init__(license_plate, brand, prod_year, color)
        self.__bike_type = bike_type
        self.__tank_capacity = tank_capacity = tank_capacity

    # Getter-Setter
    # Get Methods
    def getBikeType(self):
        return self.__bike_type
    def getTankCapacity(self):
        return self.__tank_capacity

    # Set Methods
    def setBikeType(self, bike_type):
        self.__bike_type = bike_type
    def setTankCapacity(self, tank_capacity):
        self.__tank_capacity = tank_capacity